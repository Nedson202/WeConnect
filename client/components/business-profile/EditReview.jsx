import React from 'react';
import PropTypes from 'prop-types';
import ReactStars from 'react-stars';

/**
 * @class EditReview
 *
 * @extends {Component}
 */
const EditReview = ({
  onChange,
  onStarClick,
  state,
  onReviewUpdate,
  reviewId,
  cancelEdit
}) => {
  const { editMessage, editRating, isLoading } = state;
  return (
    <div className="row">
      <div className="col-lg-9">
        <button className="close" data-dismiss="modal" aria-label="Close" onClick={cancelEdit}>
          <span aria-hidden="true">&times;</span>
        </button>
        <div className="form-group">
          <textarea
            value={editMessage}
            onChange={onChange}
            type="text"
            name="editMessage"
            className="form-control edit-review-textarea"
            id="control-label"
            placeholder="update review"
            rows="3"
          />
        </div>
        <ReactStars
          count={5}
          onChange={onStarClick}
          size={17}
          color1="#333333"
          color2="#ffaf00"
          value={editRating}
        />
      </div>
      <div className="col-lg-3">
        <button
          disabled={isLoading}
          className="btn btn-outline-success update-review-button"
          id="update-review"
          onClick={() => onReviewUpdate(reviewId)}
        >
          { !isLoading ? <span>Update review</span> : <span>Updating <i className="fa fa-spinner fa-spin" /></span>}
        </button>
      </div>
    </div>
  );
};

EditReview.propTypes = {
  onReviewUpdate: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  onStarClick: PropTypes.func.isRequired,
  cancelEdit: PropTypes.func.isRequired,
  state: PropTypes.object.isRequired,
  reviewId: PropTypes.number.isRequired
};

export default EditReview;
