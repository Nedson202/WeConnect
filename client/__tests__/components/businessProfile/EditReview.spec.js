import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import EditReview from '../../../components/business-profile/EditReview';
import businessData from '../../__mockData__/business';
import userData from '../../__mockData__/userData';

const { business, review } = businessData;
const { userUpdateData } = userData;
configure({ adapter: new Adapter() });

let props;
props = {
  onChange: jest.fn(() => Promise.resolve()),
  onStarClick: jest.fn(() => Promise.resolve()),
  onSubmit: jest.fn(() => Promise.resolve()),
  onReviewUpdate: jest.fn(() => Promise.resolve()),
  cancelEdit: jest.fn(() => Promise.resolve()),
  state: {
    message: review.message,
    rating: review.rating,
    isLoading: false
  },
  user: userUpdateData,
  business,
  reviewId: 1
};
const setup = () => shallow(<EditReview {...props} />);

describe('Component: EditReview when isLoading is false', () => {
  beforeEach(() => {
    global.window = {
      window: () => {},
    };
  });

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper.find('div').length).toBe(4);
    expect(wrapper.find('span').length).toBe(2);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  });
});

describe('Component: EditReview when isLoading is true', () => {
  beforeEach(() => {
    global.window = {
      window: () => {},
    };
  });

  it('should render', () => {
    props = {
      onChange: jest.fn(() => Promise.resolve()),
      onStarClick: jest.fn(() => Promise.resolve()),
      onSubmit: jest.fn(() => Promise.resolve()),
      onReviewUpdate: jest.fn(() => Promise.resolve()),
      state: {
        message: review.message,
        rating: review.rating,
        isLoading: true
      },
      user: userUpdateData,
      business
    };

    const wrapper = setup();
    expect(wrapper.find('i').length).toBe(1);
  });
});

describe('Submit function', () => {
  it('should submit business information', () => {
    const wrapper = setup();
    const updateButton = wrapper.find('#update-review');

    updateButton.simulate('click');
  });
});