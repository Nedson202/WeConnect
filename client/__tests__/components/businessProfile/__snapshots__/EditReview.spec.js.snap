// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: EditReview when isLoading is false should render 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-9"
  >
    <button
      aria-label="Close"
      className="close"
      data-dismiss="modal"
      onClick={[MockFunction]}
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
    <div
      className="form-group"
    >
      <textarea
        className="form-control edit-review-textarea"
        id="control-label"
        name="editMessage"
        onChange={[MockFunction]}
        placeholder="update review"
        rows="3"
        type="text"
      />
    </div>
    <ReactStars
      char="★"
      color1="#333333"
      color2="#ffaf00"
      count={5}
      edit={true}
      half={true}
      onChange={[MockFunction]}
      size={17}
      value={0}
    />
  </div>
  <div
    className="col-lg-3"
  >
    <button
      className="btn btn-outline-success update-review-button"
      disabled={false}
      id="update-review"
      onClick={[Function]}
    >
      <span>
        Update review
      </span>
    </button>
  </div>
</div>
`;
